## Random Forests

Random Forest is a classification algorithm that select random samples from a given training dataset. It will construct a decision tree for every training data, and resamples the data for each new several smaller, simpler decision trees called the forest of trees. The decision trees for the Random Forests choose the optimal split point for each decision. 

In [1]:
# Dependencies

%matplotlib inline
from matplotlib import pyplot as plt
from sklearn.datasets import make_classification
import numpy as np
import pandas as pd
from sklearn.linear_model import LogisticRegression
from sklearn.ensemble import RandomForestClassifier
from sklearn.model_selection import train_test_split
from sklearn.preprocessing import StandardScaler


In [2]:
# Create Data and scale the fit(train) dataset

X, y = make_classification(random_state=1, n_features=50, n_informative=5, n_redundant=0)
# X = pd.DataFrame(X)
df = pd.read_csv("Resources/lending_data.csv")
for col in df.columns:
    if df[col].dtype =='object':
        df[col] = pd.to_numeric(df[col], errors='coerce')

y = drop('loan_status', axis=1, inplace=True
X = df.drop(inplace=True)
    
X = df.drop('class')

In [4]:
clf = RandomForestClassifier(random_state=1, n_estimators=500).fit(X_train_scaled, y_train)
print(f'Training Score: {clf.score(X_train_scaled, y_train)}')
print(f'Testing Score: {clf.score(X_test_scaled, y_test)}')

Training Score: 1.0
Testing Score: 0.76


## Conclusion

The Random Forest Classifier has a perfect score for the training dataset which is a 100%. However, the score for the testing dataset is 76%, which is